integration-tests/types/action.ts

```diff
@@ -0,0 +1,66 @@
+// import App, { ActionConstraints } from '../../src/App';
+// import { MessageAction } from '../../src/types';
+import { App } from '@slack/bolt';
+
+const app = new App({ token: 'TOKEN', signingSecret: 'Signing Secret' });
+
+// calling action method with incorrect an type constraint value should not work
+// $ExpectError
+app.action({ type: 'Something wrong' }, ({ action }) => {
+  return action;
+});
+
+/*
+// Should error because message_action doesn't have type action_id
+// $ Expect Error
+app.action({ type: 'message_action', action_id: 'dafasf' }, ({ action }) => {
+    // NOT WORKING
```

Comment:
Once this PR lands, can we open an issue to investigate how we might get this test (and the last one in this file) to pass? I have an idea that might work, but its not worth blocking this PR over.

Reply:
btw, i tried out the idea for a couple mins and couldn't get it to work. doesn't mean its not possible, but would likely take more time than i have today to fully investigate.